# Исследование надежности заемщиков


**Цель исследования:** </br>
Проверить, влияет ли семейное положение и количество детей на факт погашения кредита в срок.

**Дополнительная задача:** </br>
построить модель кредитного скоринга — специальную систему, которая оценивает способность потенциального заёмщика вернуть кредит банку. 

**Описание данных**

- `chidren` — количество детей в семье;
- `days_employed` — общий трудовой стаж в днях;
- `dob_years` — возраст клиента в годах;
- `education` — уровень образование клиента;
- `education_id` — идентификатор уровня образования;
- `family_status` — семейное положение;
- `family_status_id` — идентификатор семейного положения;
- `gender` — пол клиента;
- `income_type` — тип занятости;
- `debt` — имел ли задолженность по возврату кредита;
- `total_income` — ежемесячный доход;
- `purpose` — цель получения кредита.


## Изучение общей информации о данных

Проведём импорт библиотек, получим информацию из файла.

In [1]:
import pandas as pd

In [2]:
try:
    data = pd.read_csv('data.csv')
except:
    data = pd.read_csv('https://..')

Изучим первые 20 строчек таблицы

In [3]:
data.head(20)

children  days_employed  dob_years            education  education_id  \
0          1   -8437.673028         42               высшее             0   
1          1   -4024.803754         36              среднее             1   
2          0   -5623.422610         33              Среднее             1   
3          3   -4124.747207         32              среднее             1   
4          0  340266.072047         53              среднее             1   
5          0    -926.185831         27               высшее             0   
6          0   -2879.202052         43               высшее             0   
7          0    -152.779569         50              СРЕДНЕЕ             1   
8          2   -6929.865299         35               ВЫСШЕЕ             0   
9          0   -2188.756445         41              среднее             1   
10         2   -4171.483647         36               высшее             0   
11         0    -792.701887         40              среднее             1   
12         0            NaN         65              среднее             1   
13         0   -1846.641941         54  неоконченное высшее             2   
14         0   -1844.956182         56               высшее             0   
15         1    -972.364419         26              среднее             1   
16         0   -1719.934226         35              среднее             1   
17         0   -2369.999720         33               высшее             0   
18         0  400281.136913         53              среднее             1   
19         0  -10038.818549         48              СРЕДНЕЕ             1   

       family_status  family_status_id gender income_type  debt  \
0    женат / замужем                 0      F   сотрудник     0   
1    женат / замужем                 0      F   сотрудник     0   
2    женат / замужем                 0      M   сотрудник     0   
3    женат / замужем                 0      M   сотрудник     0   
4   гражданский брак                 1      F   пенсионер     0   
5   гражданский брак                 1      M   компаньон     0   
6    женат / замужем                 0      F   компаньон     0   
7    женат / замужем                 0      M   сотрудник     0   
8   гражданский брак                 1      F   сотрудник     0   
9    женат / замужем                 0      M   сотрудник     0   
10   женат / замужем                 0      M   компаньон     0   
11   женат / замужем                 0      F   сотрудник     0   
12  гражданский брак                 1      M   пенсионер     0   
13   женат / замужем                 0      F   сотрудник     0   
14  гражданский брак                 1      F   компаньон     1   
15   женат / замужем                 0      F   сотрудник     0   
16   женат / замужем                 0      F   сотрудник     0   
17  гражданский брак                 1      M   сотрудник     0   
18    вдовец / вдова                 2      F   пенсионер     0   
19         в разводе                 3      F   сотрудник     0   

     total_income                                 purpose  
0   253875.639453                           покупка жилья  
1   112080.014102                 приобретение автомобиля  
2   145885.952297                           покупка жилья  
3   267628.550329              дополнительное образование  
4   158616.077870                         сыграть свадьбу  
5   255763.565419                           покупка жилья  
6   240525.971920                       операции с жильем  
7   135823.934197                             образование  
8    95856.832424                   на проведение свадьбы  
9   144425.938277                 покупка жилья для семьи  
10  113943.491460                    покупка недвижимости  
11   77069.234271       покупка коммерческой недвижимости  
12            NaN                         сыграть свадьбу  
13  130458.228857                 приобретение автомобиля  
14  165127.911772              покупка жилой недвижимости  
15  116820.904450 

Получим основную информацию о файле методом `info()`.

In [4]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21525 non-null  int64  
 1   days_employed     19351 non-null  float64
 2   dob_years         21525 non-null  int64  
 3   education         21525 non-null  object 
 4   education_id      21525 non-null  int64  
 5   family_status     21525 non-null  object 
 6   family_status_id  21525 non-null  int64  
 7   gender            21525 non-null  object 
 8   income_type       21525 non-null  object 
 9   debt              21525 non-null  int64  
 10  total_income      19351 non-null  float64
 11  purpose           21525 non-null  object 
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


*Обнаружены пропущенные значения, вероятно есть аномалии.*

## Предобработка данных

### Удаление пропусков

Получим количество пропущенных значений.

In [5]:
data.isna().sum()

children               0
days_employed       2174
dob_years              0
education              0
education_id           0
family_status          0
family_status_id       0
gender                 0
income_type            0
debt                   0
total_income        2174
purpose                0
dtype: int64

Обработаем столбец `total_income`, заполним пропуски медианным значением дохода по типу занятости, используя столбец `income_type`. 

In [6]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['total_income'].isna()), 'total_income'] = \
    data.loc[(data['income_type'] == t), 'total_income'].median()

### Обработка аномальных значений

Обработаем столбец `days_employed`, он содержит аномалию ввиде отрицательного количества дней трудового стажа. Для реальных данных это нормально. Заменим положительными значениями с помощью метода `abs()`.

In [7]:
data['days_employed'] = data['days_employed'].abs()

Выведем медианное значение трудового стажа для каждого типа занятости в днях.

In [8]:
data.groupby('income_type')['days_employed'].agg('median')

income_type
безработный        366413.652744
в декрете            3296.759962
госслужащий          2689.368353
компаньон            1547.382223
пенсионер          365213.306266
предприниматель       520.848083
сотрудник            1574.202821
студент               578.751554
Name: days_employed, dtype: float64

Имеются два типа с аномально большими значениями (безработные и пенсионеры). Их исправить будет сложно, да и к тому же они не пригодятся в исследовании, поэтому оставим как их есть. 

Выведем перечень уникальных значений столбца `children`.

In [9]:
data['children'].unique()

array([ 1,  0,  3,  2, -1,  4, 20,  5], dtype=int64)

Удалим аномальные значения

In [10]:
data = data[(data['children'] != -1) & (data['children'] != 20)]

Проверим результат работы

In [11]:
data['children'].unique()

array([1, 0, 3, 2, 4, 5], dtype=int64)

### Удаление пропусков (продолжение)

Заполним пропуски в столбце `days_employed` медианными значениями по каждого типа занятости `income_type`.

In [12]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['days_employed'].isna()), 'days_employed'] = \
    data.loc[(data['income_type'] == t), 'days_employed'].median()

Убедимся, что пропусков больше нет.

In [13]:
data.isna().sum()

children            0
days_employed       0
dob_years           0
education           0
education_id        0
family_status       0
family_status_id    0
gender              0
income_type         0
debt                0
total_income        0
purpose             0
dtype: int64

### Изменение типов данных

Заменим вещественный тип данных в столбце `total_income`.

In [14]:
data['total_income'] = data['total_income'].astype(int)

### Обработка дубликатов

Столбец `education` содержит разные значения в разных регистрах. По предварительному осмотру можно сказать, что имеются дубликаты. Приведем к нижнему регистру все значения, затем проверим  остальные столбцы.

In [15]:
data['education'] = data['education'].str.lower()

Изучим сколько строк-дубликатов присутствует.

In [16]:
data.duplicated().sum()

71

Удалим дубликаты

In [17]:
data = data.drop_duplicates()

### Категоризация данных

Проведём категоризацию дохода (`total_income`) на основе диапазонов, указанных ниже. Категоризованные данные поместим в столбец `total_income_category` с параметрами: 

- 0–30000 — `'E'`;
- 30001–50000 — `'D'`;
- 50001–200000 — `'C'`;
- 200001–1000000 — `'B'`;
- 1000001 и выше — `'A'`.


In [18]:
def categorize_income(income):
    try:
        if 0 <= income <= 30000:
            return 'E'
        elif 30001 <= income <= 50000:
            return 'D'
        elif 50001 <= income <= 200000:
            return 'C'
        elif 200001 <= income <= 1000000:
            return 'B'
        elif income >= 1000001:
            return 'A'
    except:
        pass

In [19]:
data['total_income_category'] = data['total_income'].apply(categorize_income)

Выведем из таблицы перечень уникальных целей взятия кредита.

In [20]:
data['purpose'].unique()

array(['покупка жилья', 'приобретение автомобиля',
       'дополнительное образование', 'сыграть свадьбу',
       'операции с жильем', 'образование', 'на проведение свадьбы',
       'покупка жилья для семьи', 'покупка недвижимости',
       'покупка коммерческой недвижимости', 'покупка жилой недвижимости',
       'строительство собственной недвижимости', 'недвижимость',
       'строительство недвижимости', 'на покупку подержанного автомобиля',
       'на покупку своего автомобиля',
       'операции с коммерческой недвижимостью',
       'строительство жилой недвижимости', 'жилье',
       'операции со своей недвижимостью', 'автомобили',
       'заняться образованием', 'сделка с подержанным автомобилем',
       'получение образования', 'автомобиль', 'свадьба',
       'получение дополнительного образования', 'покупка своего жилья',
       'операции с недвижимостью', 'получение высшего образования',
       'свой автомобиль', 'сделка с автомобилем',
       'профильное образование', 'высшее об

Проведём категоризацию столбца `purpose`, согласно следующему списку: 

- `'операции с автомобилем'`,
- `'операции с недвижимостью'`,
- `'проведение свадьбы'`,
- `'получение образования'`.

Данные поместим в `purpose_category`

In [21]:
def categorize_purpose(row):
    try:
        if 'автом' in row:
            return 'операции с автомобилем'
        elif 'жил' in row or 'недвиж' in row:
            return 'операции с недвижимостью'
        elif 'свад' in row:
            return 'проведение свадьбы'
        elif 'образов' in row:
            return 'получение образования'
    except:
        return 'нет категории'

In [22]:
data['purpose_category'] = data['purpose'].apply(categorize_purpose)

## Исследование данных

### Изучим зависимость между количеством детей и возвратом кредита в срок.

*Для упрощения обращения к таблице 'data' представлена основная информация*

In [23]:
print(data.columns)
print(data.shape)

Index(['children', 'days_employed', 'dob_years', 'education', 'education_id',
       'family_status', 'family_status_id', 'gender', 'income_type', 'debt',
       'total_income', 'purpose', 'total_income_category', 'purpose_category'],
      dtype='object')
(21331, 14)


*Создана сводная таблица для поиска зависимости между количеством детей и возврата кредита в срок.*

Рассчитали:
1. процент('debt_precent') должников в категориях с разным количеством детей в семьях;
2. процент('perс_of_total') категории от общего количества.

*1 - Для наглядного представления информации.
2 - Для понимания количественной составляющей каждой категории*

In [24]:
child_debt = data.pivot_table(index = ['children'], values = 'debt', aggfunc = ['count', 'sum'])
child_debt['debt_precent'] = round(child_debt['sum'] / child_debt['count'] * 100, 2)
child_debt['perс_of_total'] = round(child_debt['count'] / data.shape[0] * 100, 2)
child_debt

count   sum debt_precent perс_of_total
           debt  debt                           
children                                        
0         14091  1063         7.54         66.06
1          4808   444         9.23         22.54
2          2052   194         9.45          9.62
3           330    27         8.18          1.55
4            41     4         9.76          0.19
5             9     0         0.00          0.04

Рассчитана средняя разница в количестве должников между группами с 1-3 детьми в семьях и группой без детей. 
Группы 4 и 5 не были учтены из-за малого количества респондентов.

In [25]:
print(round(child_debt.loc[1:3, 'debt_precent'].mean() - child_debt.loc[0, 'debt_precent'], 2), '%')

    1.41
Name: 0, dtype: float64 %


*Посчитано процентное отношение граждан с детьми и без.*

In [26]:
print(f"Клиенты без детей: {child_debt['perс_of_total'][0]}%")
print(f"Учтённые клиенты с детьми: {round(child_debt['perс_of_total'][1:4].sum(), 2)}%")
print(f"Не учтённые клиенты с детьми: {round(child_debt['perс_of_total'][4:].sum(), 2)}%")

Клиенты без детей: 66.06%
Учтённые клиенты с детьми: 33.71%
Не учтённые клиенты с детьми: 0.23%


**Вывод:** \
Согласно этапу исследования, существует зависимость между возвратом кредита в срок гражданами с детьми(33%) в количестве от 1 до 3 и без детей(66%), граждане с детьми в среднем имеют задолжность на 1.41% больше, чем граждане без детей. 

### Изучим зависимость между семейным положением и возвратом кредита в срок

*Создали сводную таблицу для выполнения поставленной задачи.
Для упрощения восприятия проведена сортировка и создана колонка с процентным соотношением.*

In [27]:
family_debt = data.pivot_table(index = ['family_status'], values = 'debt', aggfunc = ['count', 'sum'])
family_debt['debt_precent'] = round(family_debt['sum'] / family_debt['count'] * 100, 2)
family_debt.sort_values(by = 'debt_precent', ascending = False)

count  sum debt_precent
                        debt debt             
family_status                                 
Не женат / не замужем   2796  273         9.76
гражданский брак        4134  385         9.31
женат / замужем        12261  927         7.56
в разводе               1189   84         7.06
вдовец / вдова           951   63         6.62

Посчитали среднюю разницу между категориями.

In [28]:
dt_pt = family_debt['debt_precent'].sort_values(ascending = False)
count_dt_pt = 0
for i in range(len(dt_pt) - 1):
    count_dt_pt += abs(dt_pt[i] - dt_pt[i+1])

print(f'Среднее увелечение количества должников от категории к категории {count_dt_pt/(len(dt_pt)-1):.2}%')   

Среднее увелечение количества должников от категории к категории 0.78%


C:\Users\Tolge\AppData\Local\Temp\ipykernel_4964\3798926019.py:4: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  count_dt_pt += abs(dt_pt[i] - dt_pt[i+1])


**Вывод:** \
Зависимость между семейным положением и возвратом кредита в срок имеет разницу диапазона значений, начиная от категории "вдова/вдовец" до "Не женат / не замужем" в размере 3.14%. Среднее увелечение количества должников от категории к категории 0.78%. Большая разница в проценте(2.2%) должников между группами клиентов, которые состояли в официальном браке и которые не состояли.

### Наличие зависимостьи между уровнем дохода и возвратом кредита в срок

Для решения поставленной задачи требуется категоризировать данные по уровню дохода.
Необходимо для общего представления вывести обзорную информацию

In [29]:
income = data['total_income'].sort_values()
min_income, med_income, max_income = income.min(), income.median(), income.max()
print('Минимальный уровень дохода:', min_income
     ,'\nСредний уровень дохода:', med_income
     ,'\nМаксимальный уровень дохода:', max_income
     ,'\n')


Минимальный уровень дохода: 20667 
Средний уровень дохода: 142594.0 
Максимальный уровень дохода: 2265604 



Собрали обзорную статистику по уровню зарплат из таблицы 'data'.
Провели предварительную категоризацию по случайным значениям.

In [30]:
income_name = ['interval', 'counter_man', 'precent']
income_data = []
lvl = [0, 80000, 100000, 120000, 200000, 250000, 500000, 3000000]

def lvl_income(lvl):
    for i in range(len(lvl) - 1):
        interval = f'{lvl[i]}-{lvl[i+1]}'
        counter_man = data.loc[(data['total_income'] >= lvl[i]) & (data['total_income'] <= lvl[i+1]), 'total_income'].count()
        precent = f'{counter_man / data.shape[0]:.3%}'
        string = [interval, counter_man, precent]
        if counter_man != 0:
            income_data.append(string)
    return pd.DataFrame(data = income_data, columns = income_name)   

lvl_income(lvl)    

interval  counter_man  precent
0         0-80000         2265  10.618%
1    80000-100000         2179  10.215%
2   100000-120000         2752  12.901%
3   120000-200000         9096  42.642%
4   200000-250000         2242  10.511%
5   250000-500000         2575  12.072%
6  500000-3000000          222   1.041%

Провели категоризацию данных, согласно общему представлению об уровне зарплат из открытых источников.

In [31]:
lvl = {'0':0, 'минимальный':40000, 'ниже среднего':80000, 'средний':120000
       ,'выше среднего':200000, 'высокий':250000, 'сверх высокий':500000, 'эксклюзив':3000000}
                    
def categor_income(income):
    for key, values in lvl.items():
        if income <= values:
            return key
    return 'some_err'        

data['income_lvl'] = data['total_income'].apply(categor_income)
data_income_lvl = data.pivot_table(index = ['income_lvl'], values = 'debt', aggfunc = ['sum', 'count','mean'])
data_income_lvl['percent_income'] = round(data_income_lvl['sum']['debt']/data_income_lvl['count']['debt'] * 100,2)
data_income_lvl.sort_values(by='percent_income')    

sum count      mean percent_income
              debt  debt      debt               
income_lvl                                       
эксклюзив       14   222  0.063063           6.31
сверх высокий  180  2575  0.069903           6.99
высокий        162  2242  0.072257           7.23
ниже среднего  164  2141  0.076600           7.66
минимальный     10   124  0.080645           8.06
средний        411  4931  0.083350           8.34
выше среднего  791  9096  0.086961           8.70

Составили промежуточный итог и краткие выводы.

In [32]:
conc_id = data_income_lvl['percent_income'].sort_values()
print(f'Максимальное количество должников у категории доходности: {conc_id.idxmax()} - {conc_id.max()}%')
print(f'Минимальное количество должников у категории доходности: {conc_id.idxmin()} - {conc_id.min()}%')
print(f'Медианное значение составило {conc_id.median()}%\n')
print(f'Разница между максимальным и минимальном значении в процентном содержании - {conc_id.max()-conc_id.min():.3}%')
count = 0
for i in range(len(conc_id) - 1):
   count += abs(conc_id[i] - conc_id[i+1])
print(f'Средняя разница между категориями в процентном содержании - {count/(len(conc_id)-1) :.2}%')

Максимальное количество должников у категории доходности: выше среднего - 8.7%
Минимальное количество должников у категории доходности: эксклюзив - 6.31%
Медианное значение составило 7.66%

Разница между максимальным и минимальном значении в процентном содержании - 2.39%
Средняя разница между категориями в процентном содержании - 0.4%


C:\Users\Tolge\AppData\Local\Temp\ipykernel_4964\3666824389.py:8: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  count += abs(conc_id[i] - conc_id[i+1])


**Вывод:** \
В среднем значение разности количества должников между категориями составляет порядка 0.4%, что свидетельствует о незначительной зависимости. Данные о разнице могут незначительно меняться в зависимости от порога категорий. \
Уровень ответственности клиентов по убыванию, где под №1 идёт наиболее ответственная группа с указанием диапазона заработной платы:
1. эксклюзив (от 3.000.000 и больше);
2. сверх высокий (500.000 - 3.000.000);
3. высокий (250.000 - 500.000);
4. ниже среднего (80.000 - 120.000);
5. минимальный (40.000 - 80.000);
6. средний (120.000 - 200.000);
7. выше среднего (200.000 - 250.000).

### Влияние целей кредита на его возврат в срок

Для решения поставленной задачи использовли метод создания сводных таблиц `pivot_table`.
Использовали колонку `"purpose_category"` полученную в пункте 2.6, задания 16
и значения из колонки `"debt"`

In [33]:
# Ваш код будет здесь. Вы можете создавать новые ячейки.
data_pc_d = data.pivot_table(index = ['purpose_category'], values = 'debt', aggfunc = ['sum', 'count', 'mean'])\
            .sort_values(by=('mean', 'debt'))\
            .style.format({('mean', 'debt'): '{:.2%}'})
data_pc_d


Рассчитали статитические данные

In [34]:
print(f"Разница между максимальным и минимальным показателем: {(9.35-7.26):.2}%")
print(f'Средний показатель разницы между группами в возврате кредита в срок: {((9.35-9.25)+(9.25-7.91)+(7.91-7.26))/3:.02}%')

Разница между максимальным и минимальным показателем: 2.1%
Средний показатель разницы между группами в возврате кредита в срок: 0.7%


**Вывод:** \
Полученные данные позволяют предположить, что категории "операции с недвижимостью" и "проведение свадьбы", выбирают более кредито-платежные клиенты. Разница с категориями 'операции с автомобилем' и 'получение образования' составляет порядка 2%. Полученные процентные показатели могут быть подвержены влиянию возраста и зрелости клиентов. Возможно кредиты на недвижимость оплачиваются не одним человеком, что снижает нагрузку, а кредиты на свадьбы хотя бы частично возвращаются денежными подарками.

### Возможные причины появления пропусков в исходных данных

Причиной пропуска в данных может быть человеческих фактор или технический.
Если речь идет о клиенте, то человек может просто не захотеть заполнять информацию о себе, а если речь идет о сотруднике, то причиной пропуска может стать ошибка. Иногда пропуски оставляют умышленно для заполнения спецсимволами.
Техническая ошибка может возникнуть из-за сбоя или ошибок программы.

### Объяснение, почему заполнять пропуски медианным значением — лучшее решение для количественных переменных.

В выборке, где значения могут иметь выбросы, среднее значение по серии даст большие отклонения в отличии от медианного значения.

## Общий вывод.

Семейное положение и количество детей имеет небольшую зависимость по нескольким показателям в отношении погашения кредита в срок. Результаты исследование представлены отношением клиентов, имевших задолженность по возврату кредитов(далее - должников) к общему количеству клиентов категорий и показаны в процентах:
- общее количество должников в исследовании с детьми выше на 1.41% по отношению к должникам без детей(таковых 33% от общей выборки). В общем случае можно сказать, что клиенты без детей справляются с выплатой кредита лучше, можно предположить, что это обусловлено трудностями в воспитании ребенка и непредвиденными расходами;
    
- количество должников в зависимости от семейного положения варируется между показателями 6.62% и 9.76%, разница между которыми составляет 3.14%. Прошу обратить внимание на группы в которых люди состояли в официальных отношения и те, где этого не было(гражданский брак, не женат/не замужем). Последние имеют склонность задерживать возврат кредита в срок чаще на 1.75%(между граничными показателями), что может быть обусловлено отсутвием опыта самостоятельной жизни. Отдельно стоит отметить группы "в разводе", "вдова/вдовец", они лучше остальных справляются с задачей возврата кредита в срок. 

- анализ по доходам проведён на основании собственных категорий клиентов. Уровень дохода имеет влияние на количество должников составляя прирост 0.4% между категориями. Общий показатель варируется между 6.31% и 8.7%. Люди с категориями зарплат 'средний', 'выше среднего', уровень которых варируется от 80000 - 200000 хуже всего справляются с возвратом кредита(таковых большинство, можно предположить, что это вызвано неопытностью в планировании распределения финансов), к ним же можно добавить категорию 'минимальнй', где уровень зарплат от 0 - 40000, но количество клиентов в этой группе всего 124. Стоит отметить наиболее стабильные и потенциально выгодные группы в соответствии с их рейтингом и доходом по погашению кредита в срок: 
 1. 'эксклюзив'(500000-3000000) всего 222 клиента;
 2. 'сверх высокий'(250000-500000);   
 3. 'высокий'(200000-250000).

- цели кредита изменяют показатель возврата в срок между группами(7.26-9.35%) в среднем на 0.7%. Полученные данные позволяют предположить, что категории "операции с недвижимостью" и "проведение свадьбы", выбирают более кредито-платежные клиенты, где заём может оплачиваться не одним человеком, что снижает нагрузку. Разница с категориями 'операции с автомобилем' и 'получение образования' составляет порядка 2%. Полученные процентные показатели могут быть подвержены влиянию возраста и зрелости клиентов.

Полученные данные говорят о незначительной разнице между категориями. Кроме того, те категории, где наблюдается увеличение времени погашения кредита, составляют большую или существенную часть от всей базы клиентов. 

Рекомендуется:
- Проанализировать данные по группам клиентов с доходами "средний", "выше среднего", обратить внимание на то, как долго клиенты получают свои зарплаты, был ли скачок в уровне зарплат.   
- Обратить внимание на клиентов с повышенным доходом и суммой кредита среди должников.
- Обратить внимание на группы 'гражданский брак', 'не женат/не замужем' и рассмотреть зависимость возврата кредита в срок от возраста и образования.